components/service/glean/src/main/java/mozilla/components/service/glean/Glean.kt

```diff
@@ -146,15 +146,25 @@ open class GleanInternalAPI {
 
     /**
      * Collect and assemble the ping. Asynchronously submits the assembled
-     * payload to the designated server using [httpPingUploader].
+     * payload to the designated server using [httpPingUploader] but only
+     * if metrics are enabled.
      */
     internal fun sendPing(store: String, docType: String) {
-        val pingContent = pingMaker.collect(store)
-        val uuid = UUID.randomUUID()
-        val path = makePath(docType, uuid)
-        // Asynchronously perform the HTTP upload off the main thread.
-        GlobalScope.launch(Dispatchers.IO) {
-            httpPingUploader.upload(path = path, data = pingContent)
+        // Do not send ping if metrics disabled
+        if (!getMetricsEnabled()) {
+            logger.debug("Attempt to send ping [$store] but metrics disabled, aborting send.")
+            return
+        }
+
+        // Since the pingMaker.collect() function sends an empty string if there is nothing recorded
```

Comment:
nit: this comment needs to be updated since we're not returning an empty string anymore from `collect`